speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -327,6 +332,40 @@ static void OnActivate(void* /* dummy */)
     LoadDefaultSettings();
 }
 
+/// @brief Opens a file dialog and changes the button to reflect this choice.
+static void SelectFile(void* /* dummy */)
+{
+    // Opens a file dialog on Windows
+    // Create file dialog
+    HRESULT hresult = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
```

Comment:
What happens if no file is selected, is there a check? or does it select a standard black box?

Reply:
If there is no file selected in the dialog, the method has an early return; if the button isn't pressed/resolved, there is no check enforcing that.

Reply:
It is now enforced.